joomla/joomla-cms
# Reduce inline scripts

Pull Request for Issue #NULL.
### Summary of Changes

In almost all singular edit views (forms) joomla needs some javascript which is injected from php, e.g.:

``` php
JFactory::getDocument()->addScriptDeclaration('
    Joomla.submitbutton = function(task)
    {
        if (task == "profile.cancel" || document.formvalidator.isValid(document.getElementById("profile-form")))
        {
            Joomla.submitform(task, document.getElementById("profile-form"));
        }
    };
');
```

This PR is beefing up the already existing function `Joomla.submitbutton` in media/system/js/core.js to incorporate the most common cases.
How it works:
In any form that uses validation (`JHtml::_('behavior.formvalidator');`) an additional class named `js-submit-button` will reproduce the code above.

Extra functionality:
- data attribute `data-cancel="view.cancel"` for forms that have a task different than `view.cancel`
- data attribute `data-skip-permissions="true"` if you don't want to process permissions validation (accelerates the process)
- data attribute `data-permissions-selector="#permissions-sliders"` specify the id of the permissions, if omitted and data-skip-permissions is set to true then `#permissions-sliders` is gonna be used

Then magic happens (not really...)
### Testing Instructions

Try all the views that are changed by this PR. Check the files tab to see which ones are modified.
### Documentation Changes Required

This behaviour needs to be documented (where?)
### B/C

This SHOULD BE 100% B/C
### Performance

Less is more...

@Fedik @andrepereiradasilva @mbabker
